src/controller/tripController.go

```diff
@@ -3,39 +3,475 @@ package controller
+	if utils.ContainsEmptyString(tripID.String()) {
+		return expenseerror.EXPENSE_INTERNAL_ERROR
+	}
+
+	// get user id from context
+	tokenUserId, ok := ctx.Value(model.ExpenseContextKeyUserID).(*uuid.UUID)
+	if !ok {
+		log.Printf("Error in tripController.DeleteTripEntry.ctx.Value(): %v", ok)
+		return expenseerror.EXPENSE_INTERNAL_ERROR
+	}
+
+	// check if trip exists
+	checkTripQueryString := "SELECT COUNT(*) FROM trip WHERE id = $1"
+	row := tc.DatabaseMgr.ExecuteQueryRow(checkTripQueryString, tripID)
+	var count int
+	if err := row.Scan(&count); err != nil {
+		log.Printf("Error in tripController.DeleteTripEntry.DatabaseMgr.ExecuteQueryRow(): %v", err)
+		return expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	if count == 0 {
+		return expenseerror.EXPENSE_NOT_FOUND
+	}
+
+	// check if user is part of trip
+	checkUserTripQueryString := "SELECT COUNT(*) FROM user_trip_association WHERE id_trip = $1 AND id_user = $2"
+	row = tc.DatabaseMgr.ExecuteQueryRow(checkUserTripQueryString, tripID, tokenUserId)
+	if err := row.Scan(&count); err != nil {
+		log.Printf("Error in tripController.DeleteTripEntry.DatabaseMgr.ExecuteQueryRow(): %v", err)
+		return expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	if count == 0 {
+		return expenseerror.EXPENSE_FORBIDDEN
+	}
+
+	// delete trip
+	deleteTripQueryString := "DELETE FROM trip WHERE id = $1"
+	if _, err := tc.DatabaseMgr.ExecuteStatement(deleteTripQueryString, tripID); err != nil {
+		log.Printf("Error in tripController.DeleteTripEntry.DatabaseMgr.ExecuteStatement(): %v", err)
+		return expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	return nil
+}
+
+func (tc *TripController) InviteUserToTrip(ctx context.Context, tripId *uuid.UUID, inviteUserRequest model.InviteUserRequest) (*model.TripResponse, *model.ExpenseServiceError) {
+	// Checks:
+	// 1. Check if tripId is empty
+	// 2. Check if trip exists
+	// 3. Check if tokenUser is part of trip
+	// 4. Check if user to invite is already invited to trip
+	// Then insert into user_trip_association
+	// Then return trip details
+
+	if utils.ContainsEmptyString(tripId.String()) {
+		return nil, expenseerror.EXPENSE_INTERNAL_ERROR
+	}
+
+	// Check if trip exists
+	checkTripQueryString := "SELECT COUNT(*) FROM trip WHERE id = $1"
+	row := tc.DatabaseMgr.ExecuteQueryRow(checkTripQueryString, tripId)
+	var count int
+	if err := row.Scan(&count); err != nil {
+		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteQueryRow(): %v", err)
+		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	if count == 0 {
+		return nil, expenseerror.EXPENSE_NOT_FOUND
+	}
+
+	// Get authenticated user id from context
+	tokenUserId, ok := ctx.Value(model.ExpenseContextKeyUserID).(*uuid.UUID)
+	if !ok {
+		log.Printf("Error in tripController.InviteUserToTrip.ctx.Value(): %v", ok)
+		return nil, expenseerror.EXPENSE_INTERNAL_ERROR
+	}
+
+	// Check if invited user exists
+	checkUserQueryString := "SELECT COUNT(*) FROM \"user\" WHERE username = $1 OR email = $2"
```

Comment:
Wie stellen wir uns das Invite denn vor? Das würde viele dieser Checks redundant machen, wenn wir uns das vorher definieren. Wir wollten doch die suggest-query benutzen, um sowieso nur User zu kriegen, die existieren. Da reicht doch dann hier Existenz-Check auf die UserId der einzuladenden Person.

Reply:
Wir haben zwei Mögleichkeiten, entweder wir binden uns an den SuggestUser() und können andere Leute nur über den Username einladen oder wir erlauben die freie Eingabe von Username oder E-Mail, sodass es später im Flow auch die Möglichkeit gibt, (noch) nicht registrierte Besucher Einladungen zu senden.

Reply:
Für existierende User würde ich mit SuggestUser() gehen, ist einfacher zu handhaben. Den zweiten Fall mit nicht existierenden Usern per E-Mail einladen würde an dem Punkt hier ja sowieso fehlschlagen, das müssen wir gesondert handhaben. Würde es erst beim simplen, ersten Fall belassen und das Feature mit neuen Usern in den Backlog aufnehmen.

Reply:
Ok, lassen wir es erstmal beim Username. Weiß aber immer noch nicht ob das eine Gute Idee ist, die SuggestUser() Funktion zu erzwingen.

Reply:
Ich wüsst nicht wo der Nachteil ist, so kriegst du Vorschläge on the fly und du solltest ja wissen wie der Username von deinem Kollegen ist. Du fragst ihn ja davor

Reply:
Wie stellst du dir den Workflow vor? Soll das wie beim SuggestUser auch ein hin und her geben bis der passende Username gefunden wurde?

Reply:
Wenn ein User anfängt Namen zu tippen, soll ein Request ans Backend geschickt werden, der User basierend auf dem Input vorschlägt. Sollte er `test` eingeben, soll der Endpunkt alle User in der DB zurückgeben, auf die `LIKE test%` zutrifft.